backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/log/LogOutput.java

```diff
@@ -21,13 +21,29 @@
 
 import java.io.PrintWriter;
 import java.io.StringWriter;
+import javax.annotation.Nullable;
 
 /**
  * Allow to redirect SonarLint logs to a custom output on client side
  */
 public interface LogOutput {
 
-  void log(String formattedMessage, Level level);
+  /**
+   * @deprecated please implement {@link #log(String, Level, String)} instead
+   */
+  @Deprecated(since = "10.0")
+  default void log(String formattedMessage, Level level) {
+    log(formattedMessage, level, null);
+  }
+
+  default void log(@Nullable String formattedMessage, Level level, @Nullable String stacktrace) {
```

Comment:
I was a bit surprised with this API, why not take the Throwable here instead of the stacktrace?

Reply:
I guess most clients will print the stacktrace anyway. This will save them to implement the "stacktraceToString"